druid-shell/src/backend/mac/window.rs

```diff
@@ -102,13 +102,23 @@ pub(crate) struct WindowHandle {
     /// a view. Also, this is better for hosted applications such as VST.
     nsview: WeakPtr,
     idle_queue: Weak<Mutex<Vec<IdleKind>>>,
+    parent: Option<Box<crate::WindowHandle>>,
```

Comment:
I think the `Box` is unnecessary.

Reply:
no because mac is different. It would be an infinitely big struct.

Reply:
All the others already have a reference in between the window handle and this option, mac doesnt.

Reply:
why does mac need to store parent handle though?

Reply:
To change the coordinate space relative to the parent

Reply:
Oh, I see, it's because on mac you've put the parent directly in the `WindowHandle`. Why not put it in `ViewState`, so it's more similar to the other platforms?

Reply:
I dont exactly know how to get that from the `WindowHandle`

Reply:
I can't test because I don't have a mac, but looking at `set_focused_text_field` gives a hint, I think:

```rust
// Here, self is a &WindowHandle
if let Some(view) = self.nsview.load().as_ref() {
    let state: *mut c_void = *view.get_ivar("viewState");
    let state = &mut (*(state as *mut ViewState));
}
```

Reply:
Oh yes I see, thanks! pushed the changes